Execution/executer.c

```diff
@@ -354,10 +406,135 @@ void * evaluate_expression(expr * expression)
+            exit(-1);
+        }
+        else{
+            if(rAsmt->expression->double_expression== NULL && type == jdouble){
+                fprintf(stderr, "Expression does not match ID type: jdouble\n");
+                exit(-1);
+            }
+            else if(rAsmt->expression->int_expression== NULL && type== jint){
+                fprintf(stderr, "Expression does not match ID type: jint\n");
+                exit(-1);
+            }
+            else if(rAsmt->expression->string_expression== NULL && type== jstring){
+                fprintf(stderr, "Expression does not match ID type: jstring\n");
+                exit(-1);
+            }
+            else{
+                void * retval = evaluate_expression(rAsmt->expression);
+                if(rAsmt->id->type == jstring)
+                {
+                    setGlobalVariable(rAsmt->id->id, ((string_meta *)retval)->data);
+                    free(retval);
+                }
+                else
+                {
+                    setGlobalVariable(rAsmt->id->id, retval);
+                }
+            }
+        }
+    }
+}
+
+void eval_b_stmt_list(b_stmt_list *bStmt ){
+    for(b_stmt_list * current = bStmt; current != NULL; current = current->next)
+    {
+        if(current->b_statement != NULL)
+        {
+            if(current->b_statement->expression != NULL)
+            {
+                void * retval = evaluate_expression(current->b_statement->expression);
+                if(retval != NULL)
+                {
+                    if(current->b_statement->expression->string_expression != NULL && ((string_meta *)retval)->is_intermediate)
+                    {
+                        g_string_free(((string_meta *)retval)->data, TRUE);
+                    }
+                    free(retval);
+                }
+            }
+            else if(current->b_statement->functionCall != NULL)
+            {
+                void * retval = evaluate_function(current->b_statement->functionCall);
+                if(retval != NULL)
+                {
+                    if(current->b_statement->functionCall->id->type == jf_str && ((string_meta *)retval)->is_intermediate)
+                    {
+                        g_string_free(((string_meta *)retval)->data, TRUE);
+                    }
+                    free(retval);
+                }
+            }
+            else if(current->b_statement->reassign!= NULL){
+                eval_r_asmt(current->b_statement->reassign);
+            }
+            else if(current->b_statement->ifBlock!= NULL){
+                eval_if(current->b_statement->ifBlock);
+            }
+            else if(current->b_statement->forLoop!= NULL){
+                eval_for(current->b_statement->forLoop);
+            }
+            else if(current->b_statement->whileLoop!= NULL){
+                eval_while(current->b_statement->whileLoop);
+            }
+        }
+    }
+    destroyGlobalScope();
```

Comment:
This causes an issue where it clears all global vars at the end of a block, so after a }, it basically forgets everything it had saved. Remove this line and the reassignment/global vars should be good.